### PR TITLE
Add crypto module

### PR DESCRIPTION
**Pull request by vladchicos**
Add hkdf_extract
Add hkdf_extract unittest

### CODE REVIEW

**Comment by Geod24** [comment]
Why the switch to Nix ? The action is doing a good job on its own.

File: dub.sdl [hunk 1]
```diff
@@ -5,3 +5,5 @@ copyright "Copyright © 2022, Vlăduț Chicoș"
 license "BSL-1.0"
 
 targetPath "build"
+
+dependency "openssl" version="~>3.2.2"
```
Comment (Geod24) [inline dub.sdl]:
```suggestion
dependency "openssl" version="~>3.2"
```

File: src/quic/crypto.d [hunk 2]
```diff
@@ -0,0 +1,68 @@
+module quic.crypto;
```
Comment (Geod24) [inline src/quic/crypto.d:1]: Every module should come with documentation

File: src/quic/crypto.d [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    EVP_KDF *kdf;
+    kdf = EVP_KDF_fetch(null, "hkdf", null);
+    EVP_KDF_CTX *kctx;
+    kctx = EVP_KDF_CTX_new(kdf);
+    EVP_KDF_free(kdf);
+
+    if (EVP_KDF_CTX_set_params(kctx, p) < 1)
+        handleErrors;
+    if (EVP_KDF_derive(kctx, initial_secret.ptr, 32, p) < 1)
+        handleErrors;
```
Comment (Geod24) [inline src/quic/crypto.d]: Why is every error fatal ?

Comment (vladchicos) [reply src/quic/crypto.d]: Another approach that I think of would be to print the error queue and then return an error value. Is this a better idea?

**Comment by vladchicos** [comment]
I used two different approaches for hkdf-expand and hkdf-extract. That is because in the meantime I figured out how to use EVP_PKEY_CTX related methods(compatible with openssl 1.1.1). Should I keep the one using EVP_KDF functions(added in openssl 3.0) or the other one? For the one with EVP_KDF I had to make(and fill with data) my own ossl_param structs due to the fact that deimos openssl does not bind yet to all the required methods(see this example for reference : https://wiki.openssl.org/index.php/EVP_Key_Derivation). Should I make a PR into deimos with them? Any kind of suggestion/solution is welcome :) . Thanks!

**Pull request closed without merging**